homeassistant/components/ovo_energy/sensor.py

```diff
@@ -1,26 +1,118 @@
+from homeassistant.components.sensor import (
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import DEVICE_CLASS_ENERGY, DEVICE_CLASS_MONETARY
+from homeassistant.const import (
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_MONETARY,
+    DEVICE_CLASS_TIMESTAMP,
+    ENERGY_KILO_WATT_HOUR,
+)
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import dt as dt_util
 
 from . import OVOEnergyDeviceEntity
 from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
 
 SCAN_INTERVAL = timedelta(seconds=300)
 PARALLEL_UPDATES = 4
 
+KEY_LAST_ELECTRICITY_COST: Final = "last_electricity_cost"
+KEY_LAST_GAS_COST: Final = "last_gas_cost"
+
+
+@dataclass
+class OVOEnergySensorEntityDescription(SensorEntityDescription):
+    """Class describing System Bridge sensor entities."""
+
+    value: Callable[[OVODailyUsage], StateType] = round
+
+
+SENSOR_TYPES_ELECTRICITY: tuple[OVOEnergySensorEntityDescription, ...] = (
+    OVOEnergySensorEntityDescription(
+        key="last_electricity_reading",
+        name="OVO Last Electricity Reading",
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        value=lambda usage: usage.electricity[-1].consumption,
+    ),
+    OVOEnergySensorEntityDescription(
+        key=KEY_LAST_ELECTRICITY_COST,
+        name="OVO Last Electricity Cost",
+        device_class=DEVICE_CLASS_MONETARY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+        icon="mdi:cash-multiple",
+        value=lambda usage: usage.electricity[-1].consumption,
+    ),
+    OVOEnergySensorEntityDescription(
+        key="last_electricity_start_time",
+        name="OVO Last Electricity Start Time",
+        entity_registry_enabled_default=False,
+        device_class=DEVICE_CLASS_TIMESTAMP,
+        value=lambda usage: dt_util.as_utc(usage.electricity[-1].interval.start),
+    ),
+    OVOEnergySensorEntityDescription(
+        key="last_electricity_end_time",
+        name="OVO Last Electricity End Time",
+        entity_registry_enabled_default=False,
+        device_class=DEVICE_CLASS_TIMESTAMP,
+        value=lambda usage: dt_util.as_utc(usage.electricity[-1].interval.end),
+    ),
+)
+
+SENSOR_TYPES_GAS: tuple[OVOEnergySensorEntityDescription, ...] = (
+    OVOEnergySensorEntityDescription(
+        key="last_gas_reading",
+        name="OVO Last Gas Reading",
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        icon="mdi:gas-cylinder",
```

Comment:
We do not set icon when a device_class is present

Reply:
This is has expressed in kWh. Overriding the icon in such cases is allowed